src/test/java/org/opentripplanner/openstreetmap/tagmapping/NorwayMapperTest.java

```diff
@@ -16,6 +23,110 @@ public class NorwayMapperTest {
     source.populateProperties(wps);
   }
 
+  static int[] expectedMaxspeeds = { 90, 80, 70, 60, 50, 40, 30 };
+
+  static String[] expectedHighways = {
+    "trunk",
+    "primary",
+    "secondary",
+    "tertiary",
+    "unclassified",
+    "residential",
+  };
+
+  static Stream<Arguments> createExpectedBicycleSafetyForMaxspeedCases() {
+    Double[][] expectedBicycleSafetyMatrix = {
+      { 10., 3.75, 3.75, 3.43, 2.5, 2.5, 1.83 },
+      { 10., 3.75, 3.75, 3.43, 2.5, 2.5, 1.83 },
+      { NaN, 3.43, 3.43, 2.5, 2.37, 2.37, 1.83 },
+      { NaN, 3.43, 3.43, 2.37, 2.37, 1.83, 1.83 },
+      { NaN, 3.43, 3.43, 1.83, 1.83, 1.83, 1.83 },
+      { NaN, NaN, NaN, 1.83, 1.83, 1.83, 1.83 },
+    };
+    ArrayList<Arguments> argumentsList = new ArrayList<>();
+    for (int i = 0; i < expectedHighways.length; i++) {
+      var highway = expectedHighways[i];
+      for (int j = 0; j < expectedMaxspeeds.length; j++) {
+        var expectedSafety = expectedBicycleSafetyMatrix[i][j];
+        if (!Double.isNaN(expectedSafety)) {
+          var maxspeed = expectedMaxspeeds[j];
+          var way = new OSMWithTags();
+          way.addTag("highway", highway);
+          way.addTag("maxspeed", String.valueOf(maxspeed));
+          argumentsList.add(Arguments.of(way, expectedSafety));
+        }
+      }
+    }
+    return argumentsList.stream();
+  }
+
+  static Stream<Arguments> createBicycleSafetyWithoutExplicitMaxspeed() {
+    Double[] expectedBicycleSafety = { 3.75, 3.75, 3.43, 3.43, 3.43, 1.83 };
+    ArrayList<Arguments> argumentsList = new ArrayList<>();
+    for (int i = 0; i < expectedHighways.length; i++) {
+      var highway = expectedHighways[i];
+      var expectedSafety = expectedBicycleSafety[i];
+      var way = new OSMWithTags();
+      way.addTag("highway", highway);
+      argumentsList.add(Arguments.of(way, expectedSafety));
+    }
+    return argumentsList.stream();
+  }
+
+  static Stream<Arguments> createLinkRoadLikeMainCases() {
+    ArrayList<Arguments> argumentsList = new ArrayList<>();
+    for (var i = 0; i < 4; i++) {
+      var highway = expectedHighways[i];
+      for (var maxspeed : expectedMaxspeeds) {
+        var mainRoad = new OSMWithTags();
+        mainRoad.addTag("highway", highway);
+        mainRoad.addTag("maxspeed", String.valueOf(maxspeed));
+        var linkRoad = new OSMWithTags();
+        linkRoad.addTag("highway", highway.concat("_link"));
+        linkRoad.addTag("maxspeed", String.valueOf(maxspeed));
+        argumentsList.add(Arguments.of(mainRoad, linkRoad));
+      }
+    }
+    return argumentsList.stream();
+  }
+
+  static Stream<Arguments> expectedBicycleSafetyForMaxspeedCases = createExpectedBicycleSafetyForMaxspeedCases();
```

Comment:
You don't need to save the result of the method call into a variable. Junit call the method for you if you use `MethodSource`: https://ddc-java.github.io/parameterized-tests/methodsource.html

I think it also works with `List` so you don't even need to convert to a `Stream`.